### ISSUE TITLE
Scratch org password not being updated

### ISSUE THREAD

**Issue by topherlandry** [SalesforceFoundation/CumulusCI#638]
Since this change (https://github.com/SalesforceFoundation/CumulusCI/commit/fde28ca7345e9ef052dffdbebc20d5247910a261) I've not been able to get a new scratch org password generated when creating an org.

`cci org info dev` will give me the password of a previous org instead of the new one and I find that I'm not actually able to log in with the new credentials.

I've reverted this change locally and everything has gone back to working as expected.

Steps:
1. Create org.
2. Delete org.
3. Create new org.
4. Attempt to log in with provided username / password.

**Comment by cpolcyn** [comment]
These steps successfully replace an existing scratch org with a new password:

```bash
$ cci org scratch dev org_name
$ cci org info org_name
$ cci org scratch_delete org_name
$ cci org scratch dev org_name
$ cci org info org_name
```

However, if the 4th step is omitted (`cci org scratch dev org_name`) then the password shown is the old password which is incorrect. This behavior is the same with or without the changes in fde28ca7345e9ef052dffdbebc20d5247910a261.

So for now, please make sure to use that 4th step and you should be okay, but we need to fix it so that's not needed.

**Comment by topherlandry** [comment]
@cpolcyn I should have been more clear in my description and recreation steps.

This is specifically broken if I delete my scratch org and then run the `dev_org` flow. When the new scratch org is created, it is created with a new password, but `org info` does not display the new password.

This is still true after the changes in #639.

**Comment by topherlandry** [comment]
Based on what I'm looking at, it looks like when `BaseFlow` goes to refresh the oauth token and finds no scratch org, it runs `create_org` in `ScratchOrgConfig`. `cci org scratch dev org_name` uses `create_scratch_org` in the keychain, so it makes sense that the password generation is being handled differently.

**Comment by cpolcyn** [comment]
Thanks @topherlandry for the extra details. Can you please open a new issue for this? You can reference this issue there. I'd like to leave this issue closed since #639 fixed the issue as I understood it at the time.
